doc/spec/tour.kk.md

```diff
@@ -647,6 +647,56 @@ that are not possible for general mutable reference cells.
 [Read more about state and multiple resumptions &adown;][#sec-multi-resume]
 {.learn}
 
+### Local Mutable Vectors {#sec-vectors;}
+
+Koka provides a built-in `vector` type that is backed by a constant sized C array of boxed values.
+
+Vectors still provide an immutable interface, but are faster that lists for random access.
+
+When using vectors locally you can 'mutate' them similarly to local variables:
+
+```
+fun vectors()
+  var myvec := vector-init(10, fn(i) i)
+  myvec[5] := 42
+  myvec.map(show).join(", ").println
+```
+
+When keeping a reference to a vector, the vector will be copied on assignment, but the items in the array will be shared. In the following example, the `copy` will be a vector that shares the same values as `myvec` except at location `5`.
+
+```
+fun vectors()
+  var myvec := vector-init(10, fn(i) i)
+  val copy = myvec
+  myvec[5] := 42
+```
+
+When updating a single item in the vector, the old vector will still contain a reference to the value, this will prevent reusing the memory of the old item.
```

Comment:
This sentence is unmotivated. I would write it as: "Koka can even reuse the memory of the elements of the vector if there is only one reference to the element. However, when using the update syntax above, Koka does not release the element from the vector in time for the in-place reuse to occur. If you want to update an element with the opportunity for in-place reuse, you can use the `update` function:" (cross-link the section on reuse, cut the first `reuse()` function)

Reply:
Thanks, this reads a lot better.